crmsh/idmgmt.py

```diff
@@ -65,6 +65,7 @@ def new(node, pfx):
             node_id = pfx
     if is_used(node_id):
         node_id = _gen_free_id(node_id)
+    node_id = re.sub(r'#', '.', node_id)
```

Comment:
1. add comments to explain why we need this substitution.
2. `str.replace()` is enough. No need to use regex.

```suggestion
    # The ID type in XML allows only `^[_a-zA-Z][\w\-.]*$`
    # `crm_attribute` command replaces `#` with `.` in this case
    node_id = node_id.replace('#', '.')
```